skglm/datafits/single_task.py

```diff
@@ -5,6 +5,7 @@
 
 from skglm.datafits.base import BaseDatafit
 from skglm.utils.sparse_ops import spectral_norm
+from skglm.solvers.prox_newton import _sparse_xj_dot
```

Comment:
Thinking back, we shall put this (and also other ones) in [``utils/sparse_ops``](https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/utils/sparse_ops.py)

WDYT @mathurinm ?

Reply:
definitely makes sense if they are used by other solvers indeed

Reply:
and they are private so we can move them at will

Reply:
Issue #174  to pin it in the backlog